src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -479,9 +479,27 @@ def show_service_principal(client, identifier):
     return client.get(object_id)
 
 
-def delete_service_principal(client, identifier):
-    object_id = _resolve_service_principal(client, identifier)
-    client.delete(object_id)
+def delete_service_principal(identifier):
```

Comment:
Should we have a flag to retain these artifacts?  Is there any use case that would require that?